interp/api.go

```diff
@@ -399,11 +410,108 @@ var shellOptsTable = [...]struct {
 	{' ', "pipefail"},
 }
 
-var bashOptsTable = [...]string{
-	// sorted alphabetically by name
-	"expand_aliases",
-	"globstar",
-	"nullglob",
+var bashOptsTable = [...]bashOpt{
+	// supported flags, sorted alphabetically by name
+	{
+		name:           "expand_aliases",
+		supported:      true,
+		allow_override: true,
+	},
+	{
+		name:           "globstar",
+		supported:      true,
+		allow_override: true,
+	},
+	{
+		name:           "nullglob",
+		supported:      true,
+		allow_override: true,
+	},
+	// options that are enabled by default in Bash, sorted alphabetically by name
```

Comment:
@mvdan if you agree, I'm thinking to also display `on` for the default Bash options (that we also implement) for the `shopt` command, which means enabling interp to determine those.